### PR TITLE
akkoma: 3.11.0 -> 3.12.1

### PR DESCRIPTION
**Pull request by tcmal**
## Description of changes

Update akkoma to 3.12.1 (aka 2024.03). Changelog [here](https://akkoma.dev/AkkomaGang/akkoma/src/branch/develop/CHANGELOG.md#2024-03).
This is a security update, therefore this should probably be cherry picked to 23.11.

This update introduces the requirement for users to configure `Pleroma.Upload.base_url` explicitly (along with the media proxy base URL if needed).
Since uploads are essential functionality for most people, this patch also adds a requirement for this field to be filled out when using the NixOS module.
Technically this is breaking, so I'm not certain how this should be done.

I also added myself as a maintainer, since I already maintain an akkoma server running on Nix, and am happy to upstream my changes.
Happy to undo this if it's not wanted.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/akkoma/default.nix [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
+, file
 , writeText
 , nixosTests
 , ...
 }:
-
 beamPackages.mixRelease rec {
-  pname = "pleroma";
-  version = "3.11.0";
+  pname = "akkoma";
```
[the remark this answers is not in the document]
Comment (illdefined) [reply pkgs/servers/akkoma/default.nix:13]: It was required in the past. Cf. <https://github.com/NixOS/nixpkgs/pull/291553#issuecomment-1978271548>.

**Comment by illdefined** [comment]
I just noticed that there is a 3.12.2 release already. May I suggest we skip 3.12.1?

File: pkgs/servers/akkoma/default.nix [hunk 2]
```diff
@@ -228,8 +221,8 @@ beamPackages.mixRelease rec {
   meta = with lib; {
     description = "ActivityPub microblogging server";
     homepage = "https://akkoma.social";
-    license = licenses.agpl3Only;
-    maintainers = with maintainers; [ mvs ];
+    license = licenses.agpl3;
```
[the remark this answers is not in the document]
Comment (illdefined) [reply pkgs/servers/akkoma/default.nix:224]: I agree.

File: pkgs/servers/akkoma/default.nix [hunk 3]
```diff
@@ -33,187 +35,178 @@ beamPackages.mixRelease rec {
...
+        mimeTypePatchPhase = ''
+          mkdir -p config
+          cat >> config/config.exs <<EOF
+          Mix.Config.config :mime, :types, %{
+            "application/xml" => ["xml"],
+            "application/xrd+xml" => ["xrd+xml"],
+            "application/jrd+json" => ["jrd+json"],
+            "application/activity+json" => ["activity+json"],
+            "application/ld+json" => ["activity+json"],
+            "image/apng" => ["apng"]
```
Comment (illdefined) [inline pkgs/servers/akkoma/default.nix:49]: May I ask why are specifically including the `image/apng` MIME type?

Comment (tcmal) [reply pkgs/servers/akkoma/default.nix:49]: This was added upstream to resolve an issue w/ misskey [here](https://akkoma.dev/AkkomaGang/akkoma/issues/657).

**Comment by tcmal** [comment]
Have (I think) set them to a default for versions up to 23.05, and required past that.

File: pkgs/servers/akkoma/default.nix (hunk 3 above)
Comment (illdefined) [reply pkgs/servers/akkoma/default.nix:49]: Thanks. I was unable to find that in my local repository checkout, but it turns out I was looking at the wrong branch.

**Review by illdefined — APPROVED** [review]

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -19427,6 +19427,12 @@
     githubId = 2389333;
     name = "Andy Tockman";
   };
+  tcmal = {
+    email = "me@aria.rip";
+    github = "tcmal";
+    githubId = "4183876";
```
Comment (illdefined) [inline maintainers/maintainer-list.nix:19433]: You may want to make this an integer instead of a string.

**Review by illdefined — CHANGES_REQUESTED** [review]
The changes to the package and module look good to me, but there is a small error in the addition to the maintainer list.

**Review by illdefined — APPROVED** [review]

**Review by LeSuisse — CHANGES_REQUESTED** [review]
`nixosTests.akkoma` and `nixosTests.akkoma-confined` seem to be failing

**Comment by illdefined** [comment]
@tcmal It looks like you omitted your earlier changes to the module test from the PR, which means that the upload base URL is no longer being set.

**Comment by tcmal** [comment]
My bad.

**Review by illdefined — APPROVED** [review]
Successfully ran the module tests locally.

**Review by LeSuisse — APPROVED** [review]

**Review by LeSuisse — COMMENTED** [review]
It looks like `nixosTests.akkoma-confined` is still broken (`nixosTests.akkoma` succeeds now)

**Review by LeSuisse — CHANGES_REQUESTED** [review]
It looks like `nixosTests.akkoma-confined` is still broken (`nixosTests.akkoma` succeeds now)

**Comment by illdefined** [comment]
> It looks like `nixosTests.akkoma-confined` is still broken (`nixosTests.akkoma` succeeds now)

The tests both pass successfully on my local machine (Linux x86-64, commit ID `f8b9215e`).

**Review by LeSuisse — APPROVED** [review]
Indeed the test seems to be a bit flaky but it's also the case on `master` so let's merge.

**Pull request merged**